packages/crypto/src/libsodium.ts

```diff
@@ -115,34 +109,28 @@ export const xchacha20NonceLength = 24;
 
 export class Xchacha20poly1305Ietf {
   public static async encrypt(message: Uint8Array, key: Uint8Array, nonce: Uint8Array): Promise<Uint8Array> {
-    await sodium.ready;
+    const additionalAuthenticatedData = undefined;
 
-    const additionalData = null;
+    const cipher = xchacha20poly1305(key, nonce, additionalAuthenticatedData);
 
-    return sodium.crypto_aead_xchacha20poly1305_ietf_encrypt(
-      message,
-      additionalData,
-      null, // secret nonce: unused and should be null (https://download.libsodium.org/doc/secret-key_cryptography/aead/chacha20-poly1305/xchacha20-poly1305_construction)
-      nonce,
-      key,
-    );
+    let ciphertext = new Uint8Array(message.length + 16);
+    ciphertext = cipher.encrypt(message, ciphertext);
+
+    return ciphertext;
```

Comment:
I think we don't need to pass an output object here. It is optional and will be created internally if undefined.

```suggestion
    return cipher.encrypt(message);
```